src/DynamoCore/Graph/Nodes/NodeInputData.cs

```diff
@@ -0,0 +1,144 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dynamo.Graph.Nodes
+{
+    public enum NodeInputTypes
+    {
+        numberInput, booleanInput, stringInput, colorInput, dateInput, selectionInput
+    };
+
+
+
+    /// <summary>
+    /// Represents a node which acts as a UI input for the graph
+    /// - may also hold a value for that input
+    /// </summary>
+    public class NodeInputData
```

Comment:
http://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonPropertyAttribute.htm
We should mark as many of these as we can as NullValueHandling = NullValueHandling.Ignore so that they don't show up in the JSON.